ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -93,17 +96,31 @@
    *                  {@code false} otherwise
    */
   StoreFindToken(Offset offset, UUID sessionId, UUID incarnationId, boolean inclusive) {
-    this(Type.JournalBased, offset, null, sessionId, incarnationId, inclusive);
+    this(Type.JournalBased, offset, null, sessionId, incarnationId, inclusive, CURRENT_VERSION);
   }
 
-  private StoreFindToken(Type type, Offset offset, StoreKey key, UUID sessionId, UUID incarnationId,
-      boolean inclusive) {
+  /**
+   * Instantiating {@link StoreFindToken}
+   * @param type the {@link Type} of the token
+   * @param offset the offset that this token refers to
+   * @param key The {@link StoreKey} that the token refers to
+   * @param sessionId the sessionId of the store that this token refers to
+   * @param incarnationId the incarnationId of the store that this token refers to
+   * @param inclusive {@code true} if the offset is inclusive or in other words the blob at the given offset is inclusive.
+   *                  {@code false} otherwise
+   * @param version refers to the version of the token
+   */
+  private StoreFindToken(Type type, Offset offset, StoreKey key, UUID sessionId, UUID incarnationId, boolean inclusive,
+      short version) {
     if (!type.equals(Type.Uninitialized)) {
       if (offset == null || sessionId == null) {
         throw new IllegalArgumentException("Offset [" + offset + "] or SessionId [" + sessionId + "] cannot be null");
       } else if (type.equals(Type.IndexBased) && key == null) {
         throw new IllegalArgumentException("StoreKey cannot be null for an index based token");
       }
+      if (version >= VERSION_2 && incarnationId == null) {
```

Comment:
Let us make this check `==`, (> VERSION_2 by itself is an illegal argument really).